### ISSUE TITLE
Add multicore option to deduplicate_bismark

### ISSUE THREAD

**Issue by insectopalo** [FelixKrueger/Bismark#364]
There are several call to samtools in the deduplicate_bismark script that would benefit from the -@ option that samtools has.

**Comment by FelixKrueger** [comment]
Hi @insectopalo 

This is true and could almost certainly be used for single-end alignments, but I am not sure whether it would work for paired-end alignments, as we need to guarantee that the script processes two consecutive lines at a time. I think this could would require some testing once I am back from holidays.

**Comment by FelixKrueger** [comment]
Sorry for the long delay, I have now run a few experiments regarding this issues, by adding additional threads to `samtools view`, both to the reading and writing file handles. I monitored the CPU usage via `top`,  and also measured the total time it took to process ~20 million single-end reads (on our head node...).

I am afraid the CPU usage for both reading and writing seem to be mostly well below 100%, irrespective of wether I used 1, 2 or 4 cores. Therefore, it doesn't seem like allowing Samtools to use more CPU system resources will result in a noteworthy performance increase.

To be perfectly honest, I'd therefore be tempted to just leave it as it is, unless you disagree vehemently?

**1 core:**
3m5.778s
3m1.591s
3m5.924s

**2 cores:**
2m36.280s
2m35.216s
2m37.697s

**4 cores:**
2m44.165s
2m37.788s
2m31.628s
